src/constructs/acm/certificate.ts

```diff
@@ -0,0 +1,43 @@
+import { Certificate, CertificateValidation } from "@aws-cdk/aws-certificatemanager";
+import type { CertificateProps } from "@aws-cdk/aws-certificatemanager/lib/certificate";
+import { HostedZone } from "@aws-cdk/aws-route53";
+import { RemovalPolicy } from "@aws-cdk/core";
+import { Stage } from "../../constants";
+import type { GuStack } from "../core";
+import { GuMigratingResource } from "../core/migrating";
+import type { GuStatefulConstruct } from "../core/migrating";
+
+type GuCertificateProps = Record<Stage, GuDnsValidatedCertificateProps> & GuMigratingResource;
```

Comment:
This is my first attempt at using https://github.com/guardian/cdk/pull/364 - hopefully this is roughly what you were expecting @akash1810!

Reply:
Took a second to parse that, but yep looks correct 👍🏽 . You might want to add [`AppIdentity`](https://github.com/guardian/cdk/blob/e4a8dfa0faa765543e96b8fd28739be881f9132d/src/constructs/core/identity.ts#L9) too so we can add the `app` tag.